File: app/src/main/java/ru/mipt/customviewsample/view/TallyCounter.java

package ru.mipt.customviewsample.view;

/**
 * An interface for a view implementing a tally counter.
 */
public interface TallyCounter {

    /**
     * Reset the counter.
     */
    void reset();

    /**
     * Increment the counter.
     */
    void increment();

    /**
     * @return The current count of the counter.
     */
    int getCount();

    /**
     * Set the counter value.
     */
    void setCount(int count);

}

File: app/src/main/java/ru/mipt/customviewsample/view/TallyCounter.kt
package ru.mipt.customviewsample.view

/**
 * An interface for a view implementing a tally counter.
 */
interface TallyCounter {

    /**
     * Reset the counter.
     */
    fun reset()

    /**
     * Increment the counter.
     */
    fun increment();

    /**
     * @return The current count of the counter.
     */
    fun getCount() : Int

    /**
     * Set the counter value.
     */
    fun setCount(count : Int)
}





